internal/vercheck/vercheck.go

```diff
@@ -53,26 +64,142 @@ func SelfUpdate(stdOut, stdErr io.Writer) error {
 		return usererr.New("curl or wget is required to update devbox. Please install either and try again.")
 	}
 
-	// Delete version cache. Keep this in-sync with whatever logic is in launch.sh
-	cacheDir := xdg.CacheSubpath("devbox")
-	versionCacheFile := filepath.Join(cacheDir, "latest-version")
-	_ = os.Remove(versionCacheFile)
+	// Delete current version file. This will trigger an update by the installScript.
+	if err := removeCurrentVersionFile(); err != nil {
+		return err
+	}
 
+	// Fetch the new launcher.
 	cmd := exec.Command("sh", "-c", installScript)
 	cmd.Stdout = stdOut
 	cmd.Stderr = stdErr
 	if err := cmd.Run(); err != nil {
 		return errors.WithStack(err)
 	}
 
-	fmt.Fprint(stdErr, "Latest version: ")
-	exe, err := os.Executable()
+	updated, err := triggerUpdate(stdErr)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	printSuccessMessage(stdErr, "Launcher", currentLauncherVersion(), updated.launcherVersion)
+	printSuccessMessage(stdErr, "Devbox", build.Version, updated.devboxVersion)
+
+	return nil
+}
+
+// selfUpdateDevbox will update the devbox CLI binary to the latest version.
+func selfUpdateDevbox(stdErr io.Writer) error {
+	// Delete current version file. This will trigger an update when the next devbox command is run.
+	if err := removeCurrentVersionFile(); err != nil {
+		return err
+	}
+
+	updated, err := triggerUpdate(stdErr)
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	cmd = exec.Command(exe, "version")
-	// The output of version is incidental, so just send it all to stdErr
-	cmd.Stdout = stdErr
+
+	printSuccessMessage(stdErr, "Devbox", build.Version, updated.devboxVersion)
+
+	return nil
+}
+
+type updatedVersions struct {
+	devboxVersion   string
+	launcherVersion string
+}
+
+// triggerUpdate runs `devbox version -v` and triggers an update since a new
+// version is available. It parses the output to get the new launcher and
+// devbox versions.
+func triggerUpdate(stdErr io.Writer) (*updatedVersions, error) {
+
+	exe, err := os.Executable()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	cmd := exec.Command(exe, "version", "-v")
```

Comment:
Possibly future hardening is adding `--json`

Reply:
that would be a good idea